### ISSUE TITLE
Any good strategies to run Simulation other then one in Online policy example

### ISSUE THREAD

**Issue by subhambiswas-angelone** [david-cortes/contextualbandits#56]
> # rounds are simulated from the full dataset
> def simulate_rounds(model, rewards, actions_hist, X_global, y_global, batch_st, batch_end):
>     np.random.seed(batch_st)
>     
>     ## choosing actions for this batch
>     actions_this_batch = model.predict(X_global[batch_st:batch_end, :]).astype('uint8')
>     
>     # keeping track of the sum of rewards received
>     rewards.append(y_global[np.arange(batch_st, batch_end), actions_this_batch].sum())
>     
>     # adding this batch to the history of selected actions
>     new_actions_hist = np.append(actions_hist, actions_this_batch)
>     
>     # now refitting the algorithms after observing these new rewards
>     np.random.seed(batch_st)
>     model.fit(X_global[:batch_end, :], new_actions_hist, y_global[np.arange(batch_end), new_actions_hist],
>               warm_start = True)
>     
>     return new_actions_hist
> 
> # now running all the simulation
> for i in range(int(np.floor(X.shape[0] / batch_size))):
>     batch_st = (i + 1) * batch_size
>     batch_end = (i + 2) * batch_size
>     batch_end = np.min([batch_end, X.shape[0]])
>     
>     for model in range(len(models)):
>         lst_actions[model] = simulate_rounds(models[model],
>                                              lst_rewards[model],
>                                              lst_actions[model],
>                                              X, y,
>                                              batch_st, batch_end)

This Loop of running simulation takes too much time to run in batches ,for a very large dataset . Are there any ideal case to run this?
Any strategies for this?

**Comment by david-cortes** [comment]
That's something for you to figure out.
